src/redirect.rs

```diff
@@ -51,8 +51,8 @@ fn get_next_request(mut request: Request, response: &Response) -> Option<Request
         response
             .headers()
             .get(LOCATION)
-            .and_then(|location| location.to_str().ok())
-            .and_then(|location| request.url().join(location).ok())
+            .and_then(|location| String::from_utf8(location.as_bytes().to_vec()).ok())
+            .and_then(|location| request.url().join(&location).ok())
```

Comment:
I'm realizing now that if decoding or joining fails we should `log::warn!()` (after https://github.com/ducaale/xh/pull/371). We can pick that up in another PR though.